File: CPP/Display.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"  // Mengimpor Komponen langsung

using namespace std;

class Display : public Komponen {
private:
    string tipe_display;  // Tipe display (misalnya LED, OLED, LCD)
    string resolusi;      // Resolusi layar (misalnya "1920x1080")
    float ukuran;         // Ukuran layar dalam inci
public:
    // Constructor default
    Display() : Komponen() {
        tipe_display = "";
        resolusi = "";
        ukuran = 0.0;
    }

    // Constructor dengan parameter
    Display(string merk, string nama, int harga, string tipe_display, string resolusi, float ukuran)
        : Komponen(merk, nama, harga) {  // Memanggil constructor dari Komponen
        this->tipe_display = tipe_display;
        this->resolusi = resolusi;
        this->ukuran = ukuran;
    }

    // Setter untuk tipe_display
    void setTipeDisplay(string tipe_display) {
        this->tipe_display = tipe_display;
    }

    // Setter untuk resolusi
    void setResolusi(string resolusi) {
        this->resolusi = resolusi;
    }

    // Setter untuk ukuran
    void setUkuran(float ukuran) {
        this->ukuran = ukuran;
    }

    // Getter untuk tipe_display
    string getTipeDisplay() {
        return tipe_display;
    }

    // Getter untuk resolusi
    string getResolusi() {
        return resolusi;
    }

    // Getter untuk ukuran
    float getUkuran() {
        return ukuran;
    }

    // Destructor
    ~Display() {
        // Destructor kosong, karena tidak ada alokasi memori dinamis
    }
};

File: PYTHON/Display.py
from Komponen import  Komponen

class Display(Komponen):
    def __init__(self, merk='', nama='', harga=0, tipe_display='', resolusi='', ukuran=0.0):
        super().__init__(merk, nama, harga)  # Memanggil constructor dari Komponen
        self.tipe_display = tipe_display  # Tipe display (misalnya LED, OLED, LCD)
        self.resolusi = resolusi  # Resolusi layar (misalnya "1920x1080")
        self.ukuran = ukuran  # Ukuran layar dalam inci

    # Setter untuk tipe_display
    def set_tipe_display(self, tipe_display):
        self.tipe_display = tipe_display

    # Setter untuk resolusi
    def set_resolusi(self, resolusi):
        self.resolusi = resolusi

    # Setter untuk ukuran
    def set_ukuran(self, ukuran):
        self.ukuran = ukuran

    # Getter untuk tipe_display
    def get_tipe_display(self):
        return self.tipe_display

    # Getter untuk resolusi
    def get_resolusi(self):
        return self.resolusi

    # Getter untuk ukuran
    def get_ukuran(self):
        return self.ukuran

